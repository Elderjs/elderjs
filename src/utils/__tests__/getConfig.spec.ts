const { getDefaultConfig } = require('../validations');

const defaultConfig = getDefaultConfig();

jest.mock('cosmiconfig', () => {
  return {
    cosmiconfigSync: () => ({
      search: () => ({ config: defaultConfig }),
    }),
  };
});

describe('#getConfig', () => {
  const output = {
    $$internal: {
      clientComponents: `${process.cwd()}/public/svelte`,
      ssrComponents: `${process.cwd()}/___ELDER___/compiled`,
    },
    build: {
      numberOfWorkers: -1,
      shuffleRequests: false,
    },
    debug: {
      automagic: false,
      build: false,
      hooks: false,
      performance: false,
      shortcodes: false,
      stacks: false,
    },
    distDir: `${process.cwd()}/public`,
    rootDir: process.cwd(),
    srcDir: `${process.cwd()}/src`,
    server: {
      prefix: '',
    },
    shortcodes: {
      closePattern: '}}',
      openPattern: '{{',
    },
    hooks: {
      disable: [],
    },
    origin: '',
    plugins: {},
  };

  beforeEach(() => {
    jest.resetModules();
  });

  it('throws but is catched, fallbacks to default', () => {
    jest.mock('fs', () => ({
      readFileSync: () => {
        throw new Error();
      },
    }));
    // eslint-disable-next-line global-require
    const getConfig = require('../getConfig').default;

    expect(getConfig()).toEqual(output);
  });

  it('it accepts custom initalization options', () => {
    jest.mock('fs', () => ({
      readFileSync: () => {
        throw new Error();
      },
    }));
    // eslint-disable-next-line global-require
    const getConfig = require('../getConfig').default;

    expect(getConfig({ context: 'serverless', rootDir: 't' })).toStrictEqual(
      expect.objectContaining({
        context: 'serverless',
        distDir: `${process.cwd()}/t/public`,
        rootDir: `${process.cwd()}/t`,
        srcDir: `${process.cwd()}/t/src`,
        $$internal: {
          clientComponents: `${process.cwd()}/t/public/svelte`,
          ssrComponents: `${process.cwd()}/t/___ELDER___/compiled`,
        },
      }),
    );
  });

  it('works', () => {
    jest.mock('fs', () => ({
      readFileSync: () =>
        JSON.stringify({
          compilerOptions: {
            outDir: 'build',
          },
        }),
    }));

    // eslint-disable-next-line global-require
    const getConfig = require('../getConfig').default;

    expect(getConfig()).toEqual(output);
  });
});
