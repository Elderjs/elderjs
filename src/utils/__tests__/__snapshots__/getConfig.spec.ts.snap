// Vitest Snapshot v1

exports[`#getConfig > it accepts custom initalization options > sets the expected default 1`] = `
{
  "$$internal": {
    "clientComponents": "/Users/nickreese/repos/elderjs/elderjs/public/_elderjs/svelte",
    "distElder": "/Users/nickreese/repos/elderjs/elderjs/public/_elderjs",
    "files": {
      "all": [
        "/Users/nickreese/repos/elderjs/elderjs/src/index.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/workerBuild.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/__tests__/workerBuild.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/build/build.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/build/parseBuildPerf.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/Elder.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/bootstrap.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/bundle.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/configureWatcher.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/getFilesAndWatcher.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/getWebsocket.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/passByReferenceUtils.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/esbuild/esbuildBundler.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/esbuild/esbuildPluginSvelte.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/esbuild/esbuildWorker.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/hookEntityDefinitions.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/hookInterface.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/index.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/types.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/hydrateComponents.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/inlineSvelteComponent.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/mountComponentsInHtml.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/partialHydration.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/prepareFindSvelteComponent.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/propCompression.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/svelteComponent.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/plugins/index.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/plugins/types.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/getRollupConfig.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/rollupPlugin.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/makeDynamicPermalinkFn.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/prepareRouter.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/routes.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/types.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/shortcodes/index.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/shortcodes/types.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/Page.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/capitalizeFirstLetter.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/createReadOnlyProxy.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/externalHelpers.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/fixCircularJson.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/getConfig.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/getPluginLocations.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/getUniqueId.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/index.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/normalizePrefix.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/normalizeSnapshot.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/notProduction.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/outputStyles.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/perf.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/permalinks.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/prepareInlineShortcode.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/prepareProcessStack.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/prepareRunHook.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/prepareServer.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/prepareShortcodeParser.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/shuffleArray.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/types.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/validations.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/windowsPathFix.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/wrapPermalinkFn.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/__tests__/__snapshots__/Elder.spec.ts.snap",
        "/Users/nickreese/repos/elderjs/elderjs/src/build/__tests__/build.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/build/__tests__/parseBuildPerf.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/__tests__/Elder.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/__tests__/hookEntityDefinitions.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/__tests__/hooks.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__fixtures__/largeProp.json",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__tests__/hydrateComponents.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__tests__/inlineSvelteComponent.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__tests__/mountComponentsInHtml.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__tests__/partialHydration.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__tests__/prepareFindSvelteComponent.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/partialHydration/__tests__/propCompression.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/plugins/__tests__/emptyPluginInNodeModules.test.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/plugins/__tests__/invalidPlugin.test.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/plugins/__tests__/pluginWithAll.test.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/plugins/__tests__/plugins.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/getRollupConfig.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/rollupPlugin.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/rollupPlugin.test.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/makeDynamicPermalinkFn.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/prepareRouter.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/routes-1.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/routes-2.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/routes-3.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/routes-4.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/routes/__tests__/routes-5.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/shortcodes/__tests__/shortcodes.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/Page.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/capitalizeFirstLetter.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/createReadOnlyProxy.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/externalHelpers.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/getConfig.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/getPluginLocations.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/getUniqueId.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/index.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/normalizePrefix.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/normalizeSnapshot.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/notProduction.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/outputStyles.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/perf.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/permalinks.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/prepareInlineShortcode.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/prepareProcessStack.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/prepareRunHook.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/prepareServer.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/prepareShortcodeParser.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/replaceCircular.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/shuffleArray.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/svelteComponent.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/validations.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/windowsPathFix.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/wrapPermalinkFn.spec.ts",
        "/Users/nickreese/repos/elderjs/elderjs/src/core/__tests__/__snapshots__/Elder.spec.ts.snap",
        "/Users/nickreese/repos/elderjs/elderjs/src/hooks/__tests__/__snapshots__/hooks.spec.ts.snap",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/__snapshots__/Page.spec.ts.snap",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/__snapshots__/getConfig.spec.ts.snap",
        "/Users/nickreese/repos/elderjs/elderjs/src/utils/__tests__/__snapshots__/validations.spec.ts.snap",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/node_modules/test-external-svelte-library/package.json",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/src/components/Component.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/src/layouts/External.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/simple/src/components/One.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/simple/src/layouts/Two.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/simple/src/routes/Three.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/node_modules/test-external-svelte-library/src/index.js",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/node_modules/test-external-svelte-library/src/components/Button.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/node_modules/test-external-svelte-library/src/components/Icon.svelte",
        "/Users/nickreese/repos/elderjs/elderjs/src/rollup/__tests__/__fixtures__/external/node_modules/test-external-svelte-library/src/components/Unused.svelte",
      ],
      "client": [],
      "hooks": "",
      "publicCssFile": "",
      "routes": [],
      "server": [],
      "shortcodes": "",
    },
    "findComponent": [Function],
    "logPrefix": "[Elder.js]:",
    "production": false,
    "serverPrefix": "",
    "ssrComponents": "/Users/nickreese/repos/elderjs/elderjs/___ELDER___/compiled",
    "status": "bootstrapping",
    "watcher": {
      "_events": {},
      "_eventsCount": 0,
      "_maxListeners": undefined,
    },
    "websocket": undefined,
  },
  "build": false,
  "context": "unknown",
  "debug": {
    "build": false,
    "hooks": false,
    "performance": false,
    "shortcodes": false,
    "stacks": false,
  },
  "distDir": "/Users/nickreese/repos/elderjs/elderjs/public",
  "hooks": {
    "disable": [],
  },
  "origin": "",
  "plugins": {},
  "prefix": "",
  "rootDir": "/Users/nickreese/repos/elderjs/elderjs",
  "server": false,
  "shortcodes": {
    "closePattern": "}}",
    "openPattern": "{{",
  },
  "srcDir": "/Users/nickreese/repos/elderjs/elderjs/src",
  "version": "1.2.3",
  "worker": false,
}
`;
